src/vendor/detectElementResize.js

```diff
@@ -10,46 +10,84 @@
  * 4) Add nonce for style element.
  **/
 
-export default function createDetectElementResize(nonce) {
-  // Check `document` and `window` in case of server-side rendering
-  var _window;
-  if (typeof window !== 'undefined') {
-    _window = window;
-  } else if (typeof self !== 'undefined') {
-    _window = self;
-  } else {
-    _window = global;
-  }
+// Check `document` and `window` in case of server-side rendering
+let windowObject;
+if (typeof window !== 'undefined') {
+  windowObject = window;
+
+  // eslint-disable-next-line no-restricted-globals
+} else if (typeof self !== 'undefined') {
+  // eslint-disable-next-line no-restricted-globals
+  windowObject = self;
+} else {
+  windowObject = global;
+}
+
+let cancelFrame = null;
+let requestFrame = null;
 
-  var attachEvent = typeof document !== 'undefined' && document.attachEvent;
+const TIMEOUT_DURATION = 20;
 
+const clearTimeoutFn = windowObject.clearTimeout;
+const setTimeoutFn = windowObject.setTimeout;
+
+const cancelAnimationFrameFn =
+  windowObject.cancelAnimationFrame ||
+  windowObject.mozCancelAnimationFrame ||
+  windowObject.webkitCancelAnimationFrame;
+
+const requestAnimationFrameFn =
+  windowObject.requestAnimationFrame ||
+  windowObject.mozRequestAnimationFrame ||
+  windowObject.webkitRequestAnimationFrame;
+
+if (cancelAnimationFrameFn == null || requestAnimationFrameFn == null) {
+  // For environments that don't support animation frame,
+  // fallback to a setTimeout based approach.
+  cancelFrame = clearTimeoutFn;
+  requestFrame = function requestAnimationFrameViaSetTimeout(callback) {
+    return setTimeoutFn(callback, TIMEOUT_DURATION);
+  };
+} else {
+  // Counter intuitively, environments that support animation frames can be trickier.
```

Comment:
also assuming there’s no possibility that one of the raf/cancel functions are not null without the other also not being null, right?

Reply:
If either request or cancel function is null, we'll hit the above timeout based code path.